core/src/com/mygdx/minigolf/controller/EntityFactory.java

```diff
@@ -2,46 +2,53 @@
 
 import com.badlogic.ashley.core.Engine;
 import com.badlogic.ashley.core.Entity;
+import com.mygdx.minigolf.model.EntityType;
 import com.mygdx.minigolf.model.components.Graphical;
 import com.mygdx.minigolf.model.components.Objective;
 import com.mygdx.minigolf.model.components.Physical;
 import com.mygdx.minigolf.model.components.Player;
-
-import java.util.Arrays;
-import java.util.List;
+import com.mygdx.minigolf.model.components.PowerUpAffectable;
+import com.mygdx.minigolf.model.components.PowerUpGiver;
 
 public class EntityFactory {
 
     private final Engine engine;
-    private final List<String> validTypes;
 
     public EntityFactory(Engine engine){
         this.engine = engine;
-        this.validTypes = Arrays.asList("golfball", "hole");
     }
 
     /**
-     *
-     * @param entityType A string stating an entity type to be returned
+     * @param entityType A valid entity type
      * @return An Entity based on the argument passed
-     * @throws IllegalArgumentException if argument is not a valid entity
      */
-    public Entity createEntity(String entityType){
-        if(!validTypes.contains(entityType)){
-            throw new IllegalArgumentException();
-        }
+    public Entity createEntity(EntityType entityType){
         Entity e = new Entity();
         switch(entityType){
-            case "golfball":
+            case golfball:
                 e.add(new Player());
                 e.add(new Physical());
                 e.add(new Graphical());
-                //e.add(new PowerUpAffectable());
+                e.add(new PowerUpAffectable());
                 break;
-            case "hole":
+            case hole:
                 e.add(new Physical());
                 e.add(new Objective());
                 e.add(new Graphical());
+                break;
+            case obstacle:
+                e.add(new Physical());
+                e.add(new Graphical());
+                break;
+            case spawn:
+                e.add(new Physical());
+            case powerup:
+                e.add(new PowerUpGiver());
+                e.add(new Physical());
+                e.add(new Graphical());
+            case surface:
+                e.add(new Physical());
+                e.add(new Graphical());
```

Comment:
Er det mulig å gjøre dette mer elegant på noe vis? 
Noen tanker:
Ha en liste med komponenter for hver entitet, ala
```
 List<ComponentClass> powerupComponents = Arrays.asList(PowerupGiver, Physical, Graphical)
 ...
 case powerup:
    powerupComponent.foreach(c -> e.add(new c()))
``` 

Eller kanskje med en map?
```
Map<EntityType, List<ComponentClass>> componentsMap = ...
...
componentsMap.get(entityType).foreach(c -> e.add(new c()))
```

Vet ikke om det går an i Java da